tasks/tasks.py

```diff
@@ -185,28 +189,46 @@ class RunPytest(JobTask):
     action_name = 'run_pytest'
 
     def __init__(self, template, build_url, test_suite,
-                 timeout=constants.RUN_PYTEST_TIMEOUT, **kwargs):
+                 timeout=constants.RUN_PYTEST_TIMEOUT,
+                 topology=None, **kwargs):
         super(RunPytest, self).__init__(template, timeout=timeout, **kwargs)
         self.build_url = build_url + '/'
         self.test_suite = test_suite
 
+        if not topology:
+            topology = {'name': constants.DEFAULT_TOPOLOGY}
+
+        self.topology_name = topology.get('name')
+
     def _before(self):
-        super(RunPytest, self)._before()
```

Comment:
If you call super in this case, it will try to create a vagrantfile using the action_name instead of topology_name (lines 54 to 60) and we don't want that.

Reply:
It's fine, it's using `vagrantfile` variable. See https://github.com/freeipa/freeipa-pr-ci/pull/114/files#diff-7771668df07a2bb1307ced58ed0e0767R215 and https://github.com/freeipa/freeipa-pr-ci/pull/114/files#diff-7771668df07a2bb1307ced58ed0e0767R32, https://github.com/freeipa/freeipa-pr-ci/pull/114/files#diff-7771668df07a2bb1307ced58ed0e0767R205

Reply:
Does it make sense to have vagrantfile as a property? I don't see any external use of it

Reply:
The important part is it enables proper use of inheritance. I see no reason to try and restrict the scope or visibility of `vagrantfile`.